xbmc/filesystem/FileCache.cpp

```diff
@@ -369,19 +369,17 @@ void CFileCache::Process()
     }
     else if (iRead < 0) // Fatal error
     {
-      CLog::Log(LOGDEBUG, "{} - <{}> source read returned a fatal error! Will wait for buffer to empty", __FUNCTION__, m_sourcePath);
+      CLog::Log(LOGDEBUG, "{} - <{}> source read returned a fatal error! Will retry", __FUNCTION__, m_sourcePath);
 
-      m_pCache->EndOfInput();
-
-      while (m_pCache->WaitForData(0, 0) > 0)
+      // Wait a bit:
+      if (m_seekEvent.WaitMSec(5000))
```

Comment:
Hi, out of curiosity: Why 5 seconds? Did it still fail with a lower value? :-)

Reply:
Dunno, I just c&p-ed it from the part above. It's a long time ago I implemented that part so I can't recall why I chose 5s.

Reply:
I think I chose 5s to prevent "hammering" the underlying filesystem, but now that I think of it, this may be a little long for some scenarios. @wsnipex : What's your idea about this? Perhaps lowering it to 1 or 2 seconds?

Reply:
1 sec sounds good to me

Reply:
I opted for the comprise after all: 2 seconds....